src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -50,50 +54,69 @@
-		"<li>display name: The name of a player as displayed in the chat and messages, e.g. when including %player% in a message. This name can be changed freely and can include colour codes, and is shared among all plugins (e.g. chat plugins will use a changed name).</li>",
-		"<li>tab list name: The name of a player used in the player lists that usually opens with the tab key. Please note that this is limited to 16 characters, including colour codes which are counted as 2 characters each, and that no two players can have the same tab list name at the same time.</li>",
+		"\t<li><strong>Players</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name:</strong> The Minecraft account name of the player. Can't be changed, but 'display name' can be changed.</li>",
+		"\t\t\t<li><strong>Display Name:</strong> The name of the player that is displayed in messages. " +
+			"This name can be changed freely and can include colour codes, and is shared among all plugins (e.g. chat plugins will use the display name).</li>",
+		"\t\t</ul>",
+		"\t</li>",
+		"\t<li><strong>Entities</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name:</strong> The custom name of the entity. Can be changed. But for living entities, " +
+			"the players will have to target the entity to see its name tag. For non-living entities, the name will not be visible at all. To prevent this, use 'display name'.</li>",
+		"\t\t\t<li><strong>Display Name:</strong> The custom name of the entity. Can be changed, " +
+			"which will also enable <em>custom name visibility</em> of the entity so name tag of the entity will be visible always.</li>",
+		"\t\t</ul>",
+		"\t</li>",
+		"\t<li><strong>Items</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name and Display Name:</strong> The <em>custom</em> name of the item (not the Minecraft locale name). Can be changed.</li>",
+		"\t\t</ul>",
+		"\t</li>",
+		"\t<li><strong>Inventories</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name and Display Name:</strong> The name/title of the inventory. " +
+			"Changing name of an inventory means opening the same inventory with the same contents but with a different name to its current viewers.</li>",
+		"\t\t</ul>",
+		"\t</li>",
 		"</ul>"})
 @Examples({"on join:",
 		"	player has permission \"name.red\"",
-		"	set the player's display name to \"<red>[admin]<gold>%name of player%\"",
-		"	set the player's tablist name to \"<green>%name of player%\"",
+		"	set the player's display name to \"<red>[admin] <gold>%name of player%\"",
+		"	set the player's tab list name to \"<green>%player's name%\"",
 		"set the name of the player's tool to \"Legendary Sword of Awesomeness\""})
-@Since("1.4.6 (players' name & display name), <i>unknown</i> (player list name), 2.0 (item name), 2.2-dev20 (inventory name)")
+@Since("before 2.1, 2.2-dev20 (inventory name), INSERT VERSION (non-living entity support, changeable inventory name)")

```

Comment:
I made it `before 2.1` because it was somehow wrong and mixed up.

Reply:
I don't really think we have to specify that it supports non-living entities nor that the inventory name is changeable since these were technically bugs.

Reply:
I don't think it is a bug, also people shouldn't ask why it doesn't work for non-living entities on older versions.